helix-view/src/document.rs

```diff
@@ -511,25 +609,77 @@ impl Document {
         config_loader: Option<Arc<syntax::Loader>>,
         config: Arc<dyn DynAccess<Config>>,
     ) -> Result<Self, Error> {
-        // Open the file if it exists, otherwise assume it is a new file (and thus empty).
-        let (rope, encoding) = if path.exists() {
-            let mut file =
-                std::fs::File::open(path).context(format!("unable to open {:?}", path))?;
-            from_reader(&mut file, encoding)?
+        if config.load().editorconfig {
+            Document::open_with_cfg(&EditorConfig, path, encoding, config_loader, config)
         } else {
-            let encoding = encoding.unwrap_or(encoding::UTF_8);
-            (Rope::from(DEFAULT_LINE_ENDING.as_str()), encoding)
+            Document::open_with_cfg(&Autodetect, path, encoding, config_loader, config)
+        }
```

Comment:
So based on my previous suggestion, this block should do
```rust
let doc_config = if editorconfig {
    // editorconfig load file
} else {
    // normal load file
};
...
// other blocks in `open`
// then configure document
match doc_config {
...
```
Would this be simpler?